ethereum/src/main/kotlin/com/rarible/blockchain/scanner/ethereum/migration/ChangeLog00002.kt

```
@@ -0,0 +1,20 @@
+package com.rarible.blockchain.scanner.ethereum.migration
+
+import com.github.cloudyrock.mongock.ChangeLog
+import com.github.cloudyrock.mongock.ChangeSet
+import com.github.cloudyrock.mongock.driver.mongodb.springdata.v3.decorator.impl.MongockTemplate
+import io.changock.migration.api.annotations.NonLockGuarded
+
+@ChangeLog(order = "00002")
+class ChangeLog00002 {
+    @ChangeSet(id = "00002_dropStatusIndex", order = "00001", author = "igorbaltiyskiy-rari")
+    fun dropStatusIndex(
+        mongockTemplate: MongockTemplate,
+        @NonLockGuarded subscriberHolder: EthereumLogEventSubscriberHolder,
+    ) {
+        val uniqueCollections = subscriberHolder.subscribers.map { it.getDescriptor().collection }.toSet()
+        uniqueCollections.forEach { collection ->
+            mongockTemplate.indexOps(collection).dropIndex("status")
```

can throw exception. better to try-catch or check that index really exists

---

Re. check that the index exists: shouldn't mongock guarantee that? `ChangeLog00001` must have been executed before this one.
Re. try-catch: I don't think it's a good idea. If there's a transient error (DB is temporarily unavailable), mongock will not drop the index but record that the migration was successful. It's better to let it fail, crash the node and retry on restart. --- Here I assume that a failed migration will crash the node, but I can't check that; I would make it a part of the liveness probe, but I can't find the k8s files

---

if for some reason pod crashed during drop and the index in fact was deleted there is no chance for it to recover despite correct state